### ISSUE TITLE
Image Tile 404 - Android Only

### ISSUE THREAD

**Issue by yigitozdemir** [Leaflet/Leaflet#3437]
Hello everyone, 

Iam developing a ionic application, i use leaflet with leaflet angular directive, i successfully showed a map on screen on ios and on browser, both leaflet and ngCordova plugin working perfectly.

But there is a problem on Android, it perfectly creates page also, but when it comes to display images it says "Failed to load resource: the server responded with a status of 404 (Not Found) http://otile1.mqcdn.com/tiles/1.0.0/map/17/76095/49116.png" if you check this url manually, it works, also it works on ios, the same code.

![screen shot 2015-05-07 at 5 07 49 pm](https://cloud.githubusercontent.com/assets/6051380/7517095/a57ccf34-f4db-11e4-9d77-5fdb4dfa1f41.png)

I tried tile providers as much as i can find on internet result is always same, so its not mqcdn or openstreetmap problem.

Is there anyone show this kind of error before? Is it a bug?

**Comment by IvanSanchez** [comment]
Install chrome (or chromium) in the android phone; run chrome (or chromium) on the desktop; attach a chrome debug session from the desktop to the phone... and you should have a full developer console in which you can see details about the network errors.

Please try that and report if you find some useful information.

**Comment by mourner** [comment]
Does it only reproduce with the angular directive, and not on official Leaflet examples?

**Comment by kitolog** [comment]
Hi, i have this issue in android emulator and on android phone too
I'm using Leaflet, angular-leaflet-directive, ionic and cordova
In browser all works fine.
Lib versions:
    "ionic": "driftyco/ionic-bower#1.0.0-rc.5",
    "leaflet": "0.7.x",
    "angular-leaflet-directive": "git@github.com:tombatossals/angular-leaflet-directive.git",
    "ngCordova": "ngCordova#0.1.15-alpha"

I'm connect my device in usb debug mode and trying to debug ionic in chrome developer tools
![2015-05-08 19 49 05](https://cloud.githubusercontent.com/assets/781366/7541134/83efc32c-f5bb-11e4-9833-9d96b9b6b67d.png)

**Comment by IvanSanchez** [comment]
What happens if you try to load one of those tiles directly in your android browser?

**Comment by kitolog** [comment]
Android browser in emator and in real device shows images without problems

**Comment by kitolog** [comment]
Hi all, I found the solution for bug with images and 404:
https://github.com/calendee/ionic-leafletjs-map-demo/issues/4#issuecomment-100250409

**Comment by mourner** [comment]
Excellent! Looking at that issue, it's not on the Leaflet side and has an easy workaround. Closing.

**Comment by LearningTools** [comment]
hello as I can add maps in this project MapBox

**Comment by paulincai** [comment]
almost 3 years later and users (me) still see this in various Cordova projects. The link mentioned above is dead. Does anyone still have the solution?

**Comment by ghybs** [comment]
Hi @paulincai,

I suspect this might be a Cordova whitelist issue.
You can find resources for this on Stack Overflow, e.g. https://stackoverflow.com/questions/38563547/osm-doesnt-show-in-android-5-1-but-is-shown-in-android-4-1-1-how-i-can-resolve/38563646#38563646 and https://stackoverflow.com/questions/46955045/integrating-openstreetmaps-with-cordova/46965955#46965955

If that is not your case and/or should you need further help for your projects, please use [Stack Overflow](https://stackoverflow.com/questions/tagged/leaflet) or [GIS Stack Exchange](https://gis.stackexchange.com/questions/tagged/leaflet).

If you are _really_ sure that this is a bug in Leaflet, please open a new issue and provide all information requested in the issue template, so that we can effectively investigate and hopefully resolve it.

**Comment by paulincai** [comment]
Hi @ghybs,

I tracked this problem down to the latest NPM for leaflet and react-leaflet. I mentioned this here: 

https://github.com/PaulLeCam/react-leaflet/issues/437#issue-289474409

In a project perfectly working with 1.2.0 I updated to 1.3.0 and I stopped getting retina. It works ok if I remove "detectRetina" or set on false. This is also experienced in Chrome device simulator. It looks like the urls of retina tiles are invalid.

I hope this clarifies and let me know if you think I should still open an issue for it. I presently downgraded and everything works fine.

Paul
